akka-actor/src/main/scala/akka/actor/ActorSystem.scala

```diff
@@ -5,22 +5,58 @@
 package akka.actor
 
 import java.io.Closeable
-import java.util.concurrent.{ ConcurrentHashMap, ThreadFactory, CountDownLatch, RejectedExecutionException }
-import java.util.concurrent.atomic.{ AtomicReference }
+import java.util.concurrent.{ ConcurrentHashMap, CountDownLatch, RejectedExecutionException, ThreadFactory }
+import java.util.concurrent.atomic.AtomicReference
+
 import com.typesafe.config.{ Config, ConfigFactory }
 import akka.event._
 import akka.dispatch._
 import akka.japi.Util.immutableSeq
 import akka.actor.dungeon.ChildrenContainer
 import akka.util._
 import akka.util.Helpers.toRootLowerCase
+
 import scala.annotation.tailrec
 import scala.collection.immutable
-import scala.concurrent.duration.{ Duration }
-import scala.concurrent.{ Await, Future, Promise, ExecutionContext, ExecutionContextExecutor }
+import scala.concurrent.duration.Duration
+import scala.concurrent.{ Await, ExecutionContext, ExecutionContextExecutor, Future, Promise }
 import scala.util.{ Failure, Success, Try }
-import scala.util.control.{ NonFatal, ControlThrowable }
-import java.util.Locale
+import scala.util.control.{ ControlThrowable, NonFatal }
+import java.util.Optional
+import scala.compat.java8.OptionConverters._
+
+object ActorSystemBootstrapSettings {
+
+  /**
+   * Scala API: Create bootstrap settings needed for starting the actor system
+   * @param classLoader If no ClassLoader is given, it obtains the current ClassLoader by first inspecting the current
+   *                    threads' getContextClassLoader, then tries to walk the stack to find the callers class loader, then
+   *                    falls back to the ClassLoader associated with the ActorSystem class.
+   * @param config Configuration to use for the actor system. If no Config is given, the default reference config will be obtained from the ClassLoader.
+   * @param defaultExecutionContext If defined the ExecutionContext will be used as the default executor inside this ActorSystem.
+   *                                If no ExecutionContext is given, the system will fallback to the executor configured under "akka.actor.default-dispatcher.default-executor.fallback".
+   */
+  def apply(classLoader: Option[ClassLoader], config: Option[Config], defaultExecutionContext: Option[ExecutionContext]): ActorSystemBootstrapSettings =
```

Comment:
see my comment above about making `ActorSystemBootstrapSettings` a mandatory explicit parameter to `apply`. Then it would be useful with the empty param.